File: backend/src/main/java/com/wecp/progressive/entity/Product.java
package com.wecp.progressive.entity;

import javax.persistence.*;

@Entity
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int productId;

    @ManyToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name = "warehouseId")
    private Warehouse warehouse;
    private String productName;
    private String productDescription;
    private int quantity;
    private Long price;

    public Product() {
    }

    public Product(int productId, int warehouseId, String productName, String productDescription, int quantity, Long price) {
        this.productId = productId;
        this.warehouse.setWarehouseId(warehouseId);
        this.productName = productName;
        this.productDescription = productDescription;
        this.quantity = quantity;
        this.price = price;
    }

    public int getProductId() {
        return productId;
    }

    public void setProductId(int productId) {
        this.productId = productId;
    }

    public Warehouse getWarehouse() {
        return warehouse;
    }

    public void setWarehouse(Warehouse warehouse) {
        this.warehouse = warehouse;
    }

    public String getProductName() {
        return productName;
    }

    public void setProductName(String productName) {
        this.productName = productName;
    }

    public String getProductDescription() {
        return productDescription;
    }

    public void setProductDescription(String productDescription) {
        this.productDescription = productDescription;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    public Long getPrice() {
        return price;
    }

    public void setPrice(Long price) {
        this.price = price;
    }
}

File: frontend/src/app/supplylink/types/Product.ts
import { Warehouse } from './Warehouse';
 
export class Product {
  public productId: number;
  public warehouse: Warehouse;
  public productName: string;
  public productDescription: string;
  public quantity: number;
  public price: number;
 
  constructor(productId: number, warehouseId: Warehouse, productName: string, productDescription: string, quantity: number, price: number) {
  }
 
  public displayInfo(): void {
    console.log(`Product ID: ${this.productId}`);
    console.log(`Warehouse: ${this.warehouse}`);
    console.log(`Product name: ${this.productName}`);
    console.log(`Product description: ${this.productDescription}`);
    console.log(`quantity: ${this.quantity}`);
    console.log(`price: ${this.price}`);
  }
}